### PR TITLE
feat(auth): design pass on confirm-email page and HTML confirmation email

### PR DESCRIPTION
**Pull request by norbert-kulus-blockydevs**
## Summary

Part of #1606. Two related surfaces from #1624 (email-confirmation activation) get a design pass:

1. `ConfirmEmailStatus.tsx` (the pending/success/error states rendered on `/confirm-email/:token`) - previously baseline text and a class name reused from an unrelated component (`guest-form__demo-callout`, meant for the register page's demo-mode info banner), which also mis-laid-out the error state (its default flex-row direction placed the Alert and buttons side by side).
2. `EmailConfirmationService.sendConfirmation`'s emailed message - previously plain-text only.

## Design choices (confirm-email page)

Invoked the `frontend-design` skill before making layout/visual decisions. This is a polish pass inside an already-established design system (tokens, `Alert`, `Button`, `StatusBadge` already exist in `index.css` / `shared/ui`), so the brief here was disciplined reuse plus one deliberate new element, not a fresh visual identity:

- **State signaling**: each outcome gets a small tone-colored "seal" glyph (a filled circle using the same `--status-success-*` / `--status-error-*` tokens as `Alert`'s left border) above the heading, so the state reads before the text does. This is the one new, purpose-built element for this page.
- **Weight asymmetry between success and error is intentional**: success stays unboxed (light touch - good news doesn't need a heavy container), error stays boxed in the existing `Alert` component with `role="alert"` (the message is dynamic/server-supplied and deserves the stronger, accessible treatment). Using the same primitive for both would have flattened that distinction.
- **CTA weighting**: "Back to sign in" is the reliable, always-works action, so it's the primary (filled) button; "Try again" is a plain-ghost secondary button, since retrying replays the same single-use token and is unlikely to change the outcome except in a genuine transient-failure case.
- **Pending state** reuses the existing `StatusBadge` `pulse` treatment (already used elsewhere in the app for "live" states) instead of inventing a new spinner - one less pattern in the codebase, not more.
- **Actions container is a plain vertical stack** (`.confirm-email__actions`), deliberately generic so a future "Resend confirmation email" button (pending #1649's backend endpoint) can slot in without a redesign.
- Copy follows the interface-voice guidance in `docs/frontend-ui-style-guide.md`: specific, active-voice headings ("We couldn't confirm this link", "You're all set") instead of generic labels, in the same conversational register as the rest of the auth flow.
- All colors/spacing/radii come from existing design tokens (`--status-*`, `--space-*`, `--radius-*`); no new hex values or one-off styling. Verified in both themes since the tokens already carry light/dark pairs.

## Email template

- Checked `MailerPort`'s `EmailMessage` (`libs/core/src/users/domain/ports/mailer.port.ts`) first - it already declares an optional `html?: string` field, and both current implementers already handle it correctly: `SmtpMailerAdapter` passes `html` straight through to nodemailer alongside `text`, and `ConsoleMailerAdapter` ignores it gracefully (no crash, dev-time logging stays text-only). **No port or adapter changes were needed** - only test coverage confirming both behaviors, including the new HTML-present case.
- Added `apps/api/src/auth/templates/confirmation-email.template.ts`: a small, self-contained `renderConfirmationEmailHtml()` function producing an inline-styled, table-based HTML email (logo/wordmark, heading, short body, one prominent CTA button, plain-link fallback, expiry footer note). Deliberately kept separate from the web app's design-tokens CSS system - email clients don't load external stylesheets or support most modern CSS, so every rule is inlined per the style guide's own carve-out for this case. User-controlled text (username) is HTML-escaped before interpolation.
- Wired into `EmailConfirmationService.sendConfirmation` - both `text` and `html` are now set on the same `EmailMessage` so any mail client picks whichever it supports.

## Scope notes

- This branch was built in parallel with #1649 (the confirm-email stuck-account bug fix, not yet merged as of this PR). #1649's scope is entirely `apps/api` with no FE files, so it should not touch anything this PR changes - but this branch may need a rebase once #1649 merges into #1606.
- A "Resend confirmation email" UI affordance is intentionally **out of scope** here: #1649's resend endpoint doesn't exist yet and has no FE wiring on this branch, so there is nothing to design against. The action area is shaped to accept it later without rework.

## Test plan

- [x] `pnpm --filter @openlinker/web lint` - 0 errors (pre-existing warnings only, none in touched files)
- [x] `pnpm --filter @openlinker/web type-check` - clean
- [x] `npx vitest run src/features/auth` - 16/16 passing
- [x] `pnpm --filter @openlinker/api lint` (scoped eslint run on touched files) - 0 errors/warnings
- [x] `pnpm --filter @openlinker/api type-check` (after a full `pnpm build` to clear a pre-existing stale cross-package `.d.ts` issue unrelated to this branch) - only one pre-existing, untouched-file error remains (`registration.service.spec.ts`)
- [x] `npx jest src/auth/email-confirmation.service.spec.ts src/auth/templates src/auth/adapters/mailer.adapter.spec.ts` - 23/23 passing

### CODE REVIEW

**Comment by norbert-kulus-blockydevs** [comment]
## Tech-lead review

Reviewed against `docs/frontend-architecture.md`, `docs/frontend-ui-style-guide.md`, and `docs/engineering-standards.md`. This is a clean, well-scoped polish PR - all 8 touched files reviewed in full repo context, including the pre-PR state of every modified file.

**Summary**: Design pass on `ConfirmEmailStatus.tsx` (pending/success/error states) plus a new HTML confirmation-email template wired into `EmailConfirmationService.sendConfirmation`. Both claims in the PR description check out against the actual code, not just the description text. No architecture-boundary issues, no hardcoded colors, both themes covered, tests are meaningful. Nothing blocking.

### Verified claims

1. **The `guest-form__demo-callout` misuse was real and is genuinely fixed.** That class is `display:flex` with no `flex-direction` (defaults to row) - correct for its one other usage in `register-form.tsx` (icon span + text, 2 children meant to sit side by side), but the pre-PR error state in `ConfirmEmailStatus.tsx` nested three unrelated children in it (`Alert`, `Button`, and a `<p><Link></p>`), which the flex-row default really did lay out side by side. The new `.confirm-email` / `.confirm-email__actions` classes are explicit `flex-direction: column`, and `register-form.tsx`'s usage of `guest-form__demo-callout` is untouched by this PR, so nothing else regresses.

2. **No hardcoded colors in the touched CSS.** `index.css` additions use only `var(--status-success-*)`, `var(--status-error-*)`, `var(--text-primary)`, `var(--text-secondary)`, `var(--space-*)`, `var(--radius-pill)`. No hex/rgb literals.

3. **Light/dark parity holds** - every token used ships both a light and dark value in `index.css`; nothing new was introduced that only exists in one theme.

4. **CTA hierarchy matches the description and is a reasonable UX call.** Bare `className="button"` on the "Back to sign in" / "Continue to sign in" links resolves to the default `.button` rule (signal-orange primary), while "Try again" uses `<Button tone="ghost">` (de-emphasized). On the "should Try again even exist" question: the token is single-use and `consumeToken` is atomic, so retrying after a **confirmed** invalid/already-consumed token will deterministically fail again - but the UI can't distinguish that from a transient failure where the original request never reached the server (network drop, dropped response) and the token was never actually consumed. In that narrower case a retry can genuinely succeed. De-emphasizing rather than removing is the right call given that ambiguity, and the PR explicitly designed `.confirm-email__actions` to accept a real "Resend confirmation email" action once #1649's endpoint exists.

5. **Username is genuinely HTML-escaped**, not just claimed. `escapeHtml()` in `confirmation-email.template.ts` handles `& < > " '` and is applied to both `username` and `link` before interpolation into the `href` and body. The unit test asserts the encoded form is present and the raw `<script>` string is not - not a tautological check.

6. **`EmailMessage.html?` was NOT added by this PR** - verified by diffing `libs/core/src/users/domain/ports/mailer.port.ts` between the PR branch and its base (`1606-demo-mode-improvements`): zero diff. The optional `html?: string` field, `SmtpMailerAdapter`, and `ConsoleMailerAdapter` are all byte-identical to base. The PR's claim of "no port or adapter changes needed" is accurate; only test coverage was added.

7. **Both mailer adapters handle `html` correctly** (unchanged code, but confirmed by reading it): `SmtpMailerAdapter.sendEmail` passes `html` straight through to `transport.sendMail`, which is a no-op passthrough when `html` is undefined. `ConsoleMailerAdapter.sendEmail` never reads `message.html` at all - it logs `text` only and resolves without touching or crashing on the field.

8. **Plain-text `text` fallback is still sent alongside `html`**, not replaced - confirmed in `EmailConfirmationService.sendConfirmation`, both fields are set on the same `sendEmail()` call.

9. **Tests are meaningful, not tautological.** The FE test asserts real rendered text and a real `href` attribute per state; the template spec asserts encoded output is present and the exploit string is absent; the service spec asserts the email body actually contains the confirmation link and heading text. None just re-assert a mock was called with its own input.

### Overlap risk with #1649

**No actual conflict exists today** - #1649 has no PR yet (confirmed via `gh pr list`/`gh issue view`; it's still an open issue with no branch pushed). So there is nothing to merge-conflict with right now.

That said, "#1649 doesn't touch any FE files" is confirmed true, but "isolated from this PR" needs one caveat: #1649's own description targets the catch-and-remap logic inside **`EmailConfirmationService.confirmEmail`** (same file this PR modifies), plus a planned new resend-confirmation endpoint/method in the same service. This PR's changes are confined to `sendConfirmation` (a different method, plus one new import at the top of the file) - the two changes sit in non-overlapping regions of `email-confirmation.service.ts`, so a clean auto-merge is likely, but it is the **same file**, not a disjoint one. Low risk, not zero - worth a quick look at the diff when #1649 lands and this branch rebases, particularly around the shared import block at the top of the file.

### Minor observations (non-blocking)

- **[SUGGESTION]** `apps/api/src/auth/templates/confirmation-email.template.ts` doesn't match any file-naming pattern documented in `docs/engineering-standards.md` (no `*.template.ts` entry exists in the Naming Conventions table). Not a defect - just flagging a documentation gap if this pattern is meant to be reused for other emails (e.g. the password-reset notifier mentioned as possible future scope in #1650's issue body).
- **[SUGGESTION]** The hex color constants in `confirmation-email.template.ts` (`BRAND_COLOR`, `TEXT_COLOR`, etc.) are outside the app's design-token system by necessity (email clients can't consume CSS custom properties) - correctly called out in the file's own header comment as a deliberate exception. No action needed, just confirming this isn't an oversight.

No BLOCKING or IMPORTANT issues found. Ready to merge as-is.

File: apps/api/src/auth/templates/confirmation-email.template.ts [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+ * @module apps/api/src/auth/templates
+ */
+
+export interface ConfirmationEmailTemplateInput {
+  username: string;
+  link: string;
+  ttlHours: number;
+}
+
+const BRAND_COLOR = '#4f46e5';
```
Comment (piotrswierzy) [inline apps/api/src/auth/templates/confirmation-email.template.ts:21]: **SUGGESTION (brand consistency).** `BRAND_COLOR = '#4f46e5'` is indigo, but OpenLinker's brand accent is signal-orange (`--accent-primary`, per `.claude/rules/ui-components.md` / #775). The email CTA button and fallback-link color therefore diverge from the app's identity — a bit odd for a PR that is explicitly a design pass. Hardcoded hex is correct here (email clients can't resolve CSS tokens — good call keeping it inlined), but consider picking a hex that approximates the signal-orange accent so the confirmation email reads as the same product as the page it lands on. Non-blocking.

File: apps/web/src/features/auth/components/ConfirmEmailStatus.tsx [hunk 2]
```diff
@@ -33,28 +44,49 @@ export function ConfirmEmailStatus({ token }: ConfirmEmailStatusProps): ReactEle
 
   if (state.status === 'success') {
     return (
```
Comment (piotrswierzy) [inline apps/web/src/features/auth/components/ConfirmEmailStatus.tsx:46]: **SUGGESTION (a11y — announce timing).** `aria-live="polite"` is placed on the root of each state, but each state renders a *fresh* subtree: on the pending→success transition React unmounts the pending node and mounts a new node that already contains "You're all set". Screen readers generally do **not** announce content that is already present when a live region first enters the DOM — the region has to exist *before* its contents change. So the success announcement here is unreliable. (The error path is fine: `Alert tone="error"` renders `role="alert"`, which browsers do announce on insertion — see `shared/ui/alert.tsx`.) If reliable success announcement matters, a stable always-mounted `aria-live` wrapper around the conditional block, or `role="status"` on the success container, is the more robust pattern. Non-blocking; the visual seal + heading already convey the state for sighted users.

File: apps/api/src/auth/templates/confirmation-email.template.ts [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+ * CSS system: email clients don't load external stylesheets or support
+ * most modern CSS, so every rule here is inlined on the element it
+ * affects, and the layout uses table-friendly, widely-supported
+ * properties only. Kept in `apps/api` (not `libs/core`) — this is a
+ * rendering detail of one mailer call, not a domain concern.
+ *
+ * @module apps/api/src/auth/templates
+ */
+
+export interface ConfirmationEmailTemplateInput {
```
Comment (piotrswierzy) [inline apps/api/src/auth/templates/confirmation-email.template.ts:15]: **SUGGESTION (minor, optional).** `docs/engineering-standards.md` says types live in a sibling `*.types.ts`. For a small render-helper input interface colocated with its one pure function this is a defensible exception (it's an interface-layer rendering detail, not a domain type), and the module header already justifies the file's placement — so I wouldn't block on it. Flagging only for consistency awareness.

**Review by piotrswierzy — APPROVED** [review]
## ✅ Approve — clean, well-scoped design pass; no blocking issues

> ⚠️ **Stacked PR.** Base is `1606-demo-mode-improvements`, not `main`. Reviewed the diff against its own base (8 files, all belonging to this change — no #1606 changes leak into the diff). Will need the noted rebase once #1649 merges into #1606.

### What this PR does
Design pass on two email-confirmation surfaces from #1624:
1. **`ConfirmEmailStatus.tsx`** — pending / success / error states rebuilt with a tone-colored "seal" glyph, real headings, a proper `.confirm-email__actions` stack, and correct primitives (`StatusBadge pulse`, `Alert`, `Button`). Replaces the borrowed-from-register `guest-form__demo-callout` class that had mis-laid-out the error state.
2. **HTML confirmation email** — new `renderConfirmationEmailHtml()` (inline-styled, table-based) wired into `EmailConfirmationService.sendConfirmation` alongside the existing plain-text body.

### Overall assessment
Solid work with good boundary discipline and honest reasoning in both the PR body and the code comments. Verified against the baseline:
- **Boundaries** ✅ — template correctly kept in `apps/api` (rendering detail), not `libs/core`; no domain/national vocabulary leaks. FE dep direction `features → shared` respected. No `platformType` branching.
- **Primitive reuse** ✅ — every prop used is valid: `StatusBadge` supports `tone="info"` + `pulse`, `Button` supports `tone="ghost"`, `Alert` renders fine without `title` and yields `role="alert"` on error tone (`shared/ui/*`).
- **Tokens / CSS** ✅ — all `--status-*`, `--radius-pill`, `--text-*`, `--space-*` tokens exist in `index.css`; no hardcoded hex on the web side; bounded `/* … (#1650) */` section comment per the frontend rule. `<Link className="button">` is an established pattern and base `.button` is the filled/primary style.
- **Security** ✅ — username is HTML-escaped before interpolation (with a dedicated XSS test); the system-generated link is defensively escaped too. No secrets, no `any`, no `console.log`.
- **Migration** ✅ — no ORM entity change; correctly none needed.
- **Tests** ✅ — template spec (incl. escaping), mailer-adapter html-present + graceful-ignore cases, service spec asserting `html`, and FE component tests all updated. AAA, edge cases covered.

### Findings

**BLOCKING** — none.

**IMPORTANT** — none.

**SUGGESTION**
- `confirmation-email.template.ts:21` — `BRAND_COLOR = '#4f46e5'` (indigo) diverges from the app's signal-orange brand accent; consider a hex closer to `--accent-primary` so the email matches the page. (inline)
- `ConfirmEmailStatus.tsx:46` — `aria-live="polite"` on a freshly-mounted success subtree may not announce (live region must pre-exist its content change); error path is robust via `Alert role="alert"`. (inline)
- `confirmation-email.template.ts:15` — input interface colocated rather than in a `*.types.ts`; acceptable exception, flagged for awareness. (inline)

### Positive observations
- Clear rationale for the success-unboxed / error-boxed asymmetry and the CTA weighting — matches the actual primitive semantics.
- Correctly discovered `MailerPort` already carries optional `html`, so the change is additive with **no port/adapter edits** — just new coverage confirming both implementers' behavior.
- XSS escaping paired with a concrete negative test; nice.
- Action container deliberately shaped to accept a future "Resend" affordance without rework, and the out-of-scope call on it is the right one given #1649 isn't merged.
